Tests/SQLKitTests/Utilities.swift

```diff
@@ -19,6 +21,9 @@ final class TestDatabase: SQLDatabase {
 
 struct GenericDialect: SQLDialect {
     init() { }
+
+    var supportsIfExistsVar: Bool = true
```

Comment:
You should be able to do something like:

```swift
struct GenericDialect: SQLDialect { 
    var supportIfExists: Bool
    ...
}
```

A settable property is capable of fulfilling a protocol requirement for a get-only property. 

Then when you create a `GenericDialect`, you can do:

```swift
let testDialect = GenericDialect(supportIfExists: false)
```

In Swift 5.1, you can supposedly even do `var supportIfExists: Bool = true` and the auto-generated initializer will have a default argument. 

Since it's declared as `var` you can also mutate it as long as you have a reference to a mutable `GenericDialect` (which you do on `TestDatabase`). The only time you wouldn't be able to mutate `supportIfExists` is if you were to downcast your `GenericDialect` to a `SQLDialect`. Then it would be get-only.

Reply:
Yeah I thought that was possible at first too, but supportsIfExists is a read-only property inherited from SQLDialect, so as far as I know you can only override that with another property (like I did)

Reply:
Did you get an error with just `var supportIfExists: Bool` as a property? AFAIK that should override the default extension on `SQLDialect`. Are you sure you didn't have a reference to `SQLDialect` instead of the `GenericDialect`?

Reply:
You were right, not sure what I did wrong last time. I did have to remove the `init () { }`, since that of course overrides the auto-generated initializer.

Reply:
>In Swift 5.1, you can supposedly even do var supportIfExists: Bool = true and the auto-generated initializer will have a default argument.

I just quickly checked, and this works in Swift 5.0 too